src/pds/llm/GPT2_Embeddings.py

```diff
@@ -0,0 +1,171 @@
+from transformers import GPT2LMHeadModel, FeatureExtractionPipeline, GPT2Tokenizer
+from tokenization.pds_tokenizer import word_tokenize_pds4_xml_files
+import re
+
+
+URLS = {
+    "cassini": 'https://atmos.nmsu.edu/PDS/data/PDS4/saturn_iono/data/rss_s10_r007_ne_e.xml',
+    "insight": 'https://planetarydata.jpl.nasa.gov/img/data/nsyt/insight_cameras/data/sol/0024/mipl/edr/icc/C000M0024_598662821EDR_F0000_0558M2.xml'
+}
+tokenize = GPT2Tokenizer.from_pretrained('gpt2')
+model = GPT2LMHeadModel.from_pretrained('gpt2')
+feature_extraction_pipeline = FeatureExtractionPipeline(model=model, tokenizer=tokenize)
+
+search_terms = [
+    "saturn",
+    "saturn's rings",
+    "cassini",
+    "huygens",
+    "orbiter",
+    "rss",
+    "ionospheric",
+    "ionosphere",
+    "electron density",
+    "insight",
+    "context camera",
+    "camera",
+    "mars",
+    "image"
+]
+
+
+def clean_tokens_before_embedding(tokens):
+    pattern = r"[^a-zA-Z0-9' ]"
+    clean_tokens = [re.sub(pattern, '', token) for token in tokens]
+    clean_tokens = [token.replace("'", "").replace('"', '') for token in clean_tokens]
+    clean_tokens = [token for token in clean_tokens if token.strip()]
+    return clean_tokens
+
+def get_tokens(url):
+    tokens = word_tokenize_pds4_xml_files(url)
+    clean_tokens = clean_tokens_before_embedding(tokens)
+    return clean_tokens
+
+def get_embeddings_for_pds_labels(tokens):
+    embeddings = feature_extraction_pipeline(tokens)
+    return embeddings
+def get_embeddings_for_search_terms(search_terms):
+    search_embeddings = feature_extraction_pipeline(search_terms)
+    return search_embeddings
+def main():
+    label_embeddings = {}
+    search_term_embeddings = {}
+    for label, url in URLS.items():
+        tokens = get_tokens(url)
+        embeddings = get_embeddings_for_pds_labels(tokens)
+        label_embeddings[label] = embeddings
+
+    search_embeddings = get_embeddings_for_search_terms(search_terms)
+    for i, term in enumerate(search_terms):
+        search_term_embeddings[term] = search_embeddings[i][0]
+    print(label_embeddings)
+    print(search_term_embeddings)
+if __name__ == '__main__':
+    main()
+
+'''
```

Comment:
It looks like the rest of the file is commented-out?

When you're considering commenting-out code, I really prefer either of these approaches take place:

1. Go ahead and comment it out, but always add a comment explaining _why_. This helps reduce future users and readers of the code from suffering bewilderment.
2. Go ahead and delete it!